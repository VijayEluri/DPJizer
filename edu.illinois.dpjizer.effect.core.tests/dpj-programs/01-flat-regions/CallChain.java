/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package flatregion.callchain;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C1 {
	region r1, r3;
	int x in r1;
	C2 c2 in r3;
	
	C1() {}
	
	C2 m1() {
		x = 1;
		c2 = new C2();
		return c2;
	}
}

class C2 {
	region r2;
	int y in r2;
	
	C2() {}
	
	void m2() {
		y = 2;
	}
}

class C3 {
	
	C3() {}
	
	void m3() {
		C1 c1 = new C1();
		c1.m1().m2();
	}
}