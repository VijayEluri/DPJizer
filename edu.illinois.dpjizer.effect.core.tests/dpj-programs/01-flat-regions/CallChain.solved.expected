flatregion.callchain.C1.C1() pure
flatregion.callchain.C1.m1() writes Root : flatregion.callchain.C1.r1, Root : flatregion.callchain.C1.r3
flatregion.callchain.C2.C2() pure
flatregion.callchain.C2.m2() writes Root : flatregion.callchain.C2.r2
flatregion.callchain.C3.C3() pure
flatregion.callchain.C3.m3() writes Root : flatregion.callchain.C1.r1, Root : flatregion.callchain.C1.r3, Root : flatregion.callchain.C2.r2
